homeassistant/components/smarttub/sensor.py

```diff
@@ -2,17 +2,49 @@
 from enum import Enum
 import logging
 
+import smarttub
+import voluptuous as vol
+
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import config_validation as cv, entity_platform
+
 from .const import DOMAIN, SMARTTUB_CONTROLLER
 from .entity import SmartTubSensorBase
 
 _LOGGER = logging.getLogger(__name__)
 
+
 ATTR_DURATION = "duration"
 ATTR_LAST_UPDATED = "last_updated"
 ATTR_MODE = "mode"
 ATTR_START_HOUR = "start_hour"
 
 
+SET_PRIMARY_FILTRATION_SCHEMA = vol.Schema(
+    vol.All(
+        cv.has_at_least_one_key(ATTR_DURATION, ATTR_START_HOUR),
+        {
+            vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+            vol.Optional(ATTR_DURATION): vol.All(int, vol.Range(min=1, max=24)),
+            vol.Optional(ATTR_START_HOUR): vol.All(int, vol.Range(min=0, max=23)),
+        },
+    )
+)
+
+SET_SECONDARY_FILTRATION_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_id,
```

Comment:
```suggestion
```

Reply:
This causes my test to fail:

```
E           voluptuous.error.MultipleInvalid: extra keys not allowed @ data['entity_id']
```

should I be invoking it differently?

https://github.com/home-assistant/core/pull/46980/files#diff-cf682d19cefe9521213511541b0f809f32a5a3ad840c5098c97d3b9f6ec5cc8bR60

Reply:
If you make it a `dict` it will be passed though `make_entity_service_schema` which will add the right keys/values pairs